lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -17,18 +17,9 @@
  * @package gutenberg
  */
 
-add_action( 'switch_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'switch_theme', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'start_previewing_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'start_previewing_theme', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'upgrader_process_complete', '_wp_theme_has_theme_json_clean_cache_upon_upgrading_active_theme', 10, 2 );
-add_action( 'save_post_wp_global_styles', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'activated_plugin', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'deactivated_plugin', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'upgrader_process_complete', array( 'WP_Theme_JSON_Resolver_Gutenberg', '_clean_cached_data_upon_upgrading', 10, 2 ) );
-add_action( 'save_post_wp_global_styles', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'switch_theme', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'start_previewing_theme', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'activated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'deactivated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'upgrader_process_complete', '_gutenberg_get_global_stylesheet_clean_cache_upon_upgrading', 10, 2 );
+/**
+ * When backporting to core, the existing filters hooked to WP_Theme_JSON_Resolver::clean_cached_data()
+ * need to be removed.
+ */
+add_action( 'start_previewing_theme', '_gutenberg_clean_theme_json_caches' );
```

Comment:
I've only left the `start_previewing_theme` and `switch_theme` events as they are necessary for certain flows (preview themes, theme directory) that may access the theme data before they are dispatched.

These new events were introduced in 14.7 ([stylesheet PR](https://github.com/WordPress/gutenberg/pull/45679)) when we aimed to make this persistent and can be removed: `activated_plugin`, `deactivated_plugin`, `upgrader_process_complete`.

With this change, all caches need the same invalidation logic, so I've consolidated it into a single place as per feedback at https://github.com/WordPress/gutenberg/pull/45372#discussion_r1031748867

Reply:
If we merge https://github.com/WordPress/gutenberg/pull/45955, it will mean that no cache invalidation will be needed for start_previewing_theme / switch_theme, as the act of changing / filter the stylesheet option, will cache the cache key.

Reply:
Switching themes, mean we do not have cache to clear, as each theme would have it's own cache.

Reply:
> Switching themes, mean we do not have cache to clear, as each theme would have it's own cache.

I agree. However... resetting the cache won't be a bad thing since we'll be saving some memory 👍

Reply:
My point is that to be able to remove these hooks, we need we make changes to those three places. It cannot be done in this PR or https://github.com/WordPress/gutenberg/pull/45955 that only addresses `wp_theme_has_theme_json` as far as I could see.